src/attr/_make.py

```diff
@@ -845,6 +857,16 @@ def _create_slots_class(self):
             if k not in (*tuple(self._attr_names), "__dict__", "__weakref__")
         }
 
+        if PY_3_14_PLUS:
+            # Clean up old dict to avoid leaks.
+            old_cls_dict = self._cls.__dict__ | _deproxier
+            old_cls_dict.pop("__dict__", None)
+            if "__weakref__" in self._cls.__dict__:
+                del self._cls.__weakref__
+
+            # Manually bump internal version tag.
+            self._cls.__name__ = self._cls.__name__
```

Comment:
@JelleZijlstra so `__abstractmethods__` didn't work, because it's not omnipresent – will this work too?

Reply:
No, modifying the name doesn't bump the type version. I think if there's no `__abstractmethods__`, you can set it to a nonempty value and then `del` it.